### PR TITLE
Move BinaryTableValue to public section

### PR DESCRIPTION
**Pull request by BenBorn**
#879

## Proposed Changes

Move the BinaryTableValue class to the public section, because its returned by:
'result.BasicProperties.Headers.TryGetValue("key", out object value);'

## Types of Changes

What types of changes does your code introduce to this project?
_Put an `x` in the boxes that apply_

- [x ] Bug fix (non-breaking change which fixes issue #NNNN)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause an observable behavior change in existing systems)
- [ ] Documentation improvements (corrections, new content, etc)
- [ ] Cosmetic change (whitespace, formatting, etc)

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@BenBorn Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/rabbitmq?repositoryId=rabbitmq/rabbitmq-dotnet-client&pullRequestId=880)!

[Click here](https://cla.pivotal.io/sync/rabbitmq?repositoryId=rabbitmq/rabbitmq-dotnet-client&pullRequestId=880) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@BenBorn Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/rabbitmq?repositoryId=rabbitmq/rabbitmq-dotnet-client&pullRequestId=880)!

File: projects/RabbitMQ.Client/client/api/BinaryTableValue.cs [hunk 1]
```diff
@@ -38,7 +38,7 @@
 //  Copyright (c) 2007-2020 VMware, Inc.  All rights reserved.
 //---------------------------------------------------------------------------
 
-namespace RabbitMQ.Client.Impl
+namespace RabbitMQ.Client.client.api
```
Comment (michaelklishin) [inline projects/RabbitMQ.Client/client/api/BinaryTableValue.cs:41]: The namespace this client uses for public API bits is `RabbitMQ.Client`, not `RabbitMQ.Client.client.api`

Comment (BenBorn) [reply projects/RabbitMQ.Client/client/api/BinaryTableValue.cs]: Yes thanks missed that. Now it's fixed and only the file BinaryTablevalue is changed

**Comment by lukebakken** [comment]
@michaelklishin I'll wrap this one up this morning.

**Comment by lukebakken** [comment]
@BenBorn thanks!

**Pull request merged**